metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -925,14 +926,21 @@ ApiError electLeader(String topic, int partitionId, boolean uncleanOk,
         return ControllerResult.of(records, null);
     }
 
-    ControllerResult<Void> maybeFenceStaleBrokers() {
+    ControllerResult<Void> maybeFenceOneStaleBroker() {
         List<ApiMessageAndVersion> records = new ArrayList<>();
         BrokerHeartbeatManager heartbeatManager = clusterControl.heartbeatManager();
         List<Integer> staleBrokers = heartbeatManager.findStaleBrokers();
-        for (int brokerId : staleBrokers) {
+        if (!staleBrokers.isEmpty()) {
```

Comment:
If we are only expiring one, maybe we can change `findStaleBrokers` to `findStaleBroker` and return `Optional<Integer>` instead of `List<Integer>`.

Reply:
That is fair. Let me do that (wasn't sure if we wanted to add another method for this case).